### PR TITLE
Alfinflow 351 introduce SlingShotSpringClient

### PR DESCRIPTION
**Pull request by driesdeproost**

### CODE REVIEW

**Review by JanVdloock — APPROVED** [review]

File: alfresco-webscripts-client/alfresco-webscripts-spi/src/main/java/eu/xenit/alfresco/webscripts/client/spi/model/slingshot/Metadata.java [hunk 1]
```diff
@@ -45,13 +48,19 @@
     @AllArgsConstructor
     public static class ValueContainer {
         String dataType;
-        String value;
+        String value; // TODO Serializable instead maybe?
+        @JsonSetter("isContent")
```
Comment (kerkhofsd) [inline alfresco-webscripts-client/alfresco-webscripts-spi/src/main/java/eu/xenit/alfresco/webscripts/client/spi/model/slingshot/Metadata.java]: Why do we need this :thinking: ?

File: alfresco-webscripts-client/alfresco-webscripts-ditto/src/integration-test/java/eu/xenit/alfresco/webscripts/client/ditto/SlingShotFakeClientIntegrationTest.java [hunk 2]
```diff
@@ -19,4 +44,22 @@ public NodeLocatorClient nodeLocatorClient() {
         return new NodeLocatorFakeClient(dataSet);
     }
 
+    @Test
+    public void associationsTest() {
```
Comment (kerkhofsd) [inline alfresco-webscripts-client/alfresco-webscripts-ditto/src/integration-test/java/eu/xenit/alfresco/webscripts/client/ditto/SlingShotFakeClientIntegrationTest.java:48]: I don't like that we have different tests for different implementations. Nevertheless I get why you needed to do this here. Having a look at possible improvements.

File: alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/SlingShotSpringClient.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package eu.xenit.alfresco.webscripts.client.spring;
+
+import eu.xenit.alfresco.webscripts.client.spi.SlingShotClient;
+import eu.xenit.alfresco.webscripts.client.spi.model.slingshot.Metadata;
+import java.net.URI;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+public class SlingShotSpringClient implements SlingShotClient {
```
Comment (kerkhofsd) [inline alfresco-webscripts-client/alfresco-webscripts-spring/src/main/java/eu/xenit/alfresco/webscripts/client/spring/SlingShotSpringClient.java:9]: :+1: for using "conventions"

File: alfresco-webscripts-client/alfresco-webscripts-spi/src/main/java/eu/xenit/alfresco/webscripts/client/spi/model/slingshot/Metadata.java (hunk 1 above)
Comment (driesdeproost) [reply alfresco-webscripts-client/alfresco-webscripts-spi/src/main/java/eu/xenit/alfresco/webscripts/client/spi/model/slingshot/Metadata.java]: It won't recognize the field for deserialization without it. 
I guess it had something to do with the lombok `@Data`

I'll try to reproduce

File: alfresco-webscripts-client/alfresco-webscripts-ditto/src/integration-test/java/eu/xenit/alfresco/webscripts/client/ditto/SlingShotFakeClientIntegrationTest.java (hunk 2 above)
Comment (driesdeproost) [reply alfresco-webscripts-client/alfresco-webscripts-ditto/src/integration-test/java/eu/xenit/alfresco/webscripts/client/ditto/SlingShotFakeClientIntegrationTest.java:48]: We could put documents with association in the bootstrapped Alfresco in ditto?

**Pull request merged**